regression: sysbuild with encrypted image fails to build

### Describe the bug

In v4.3.0 this build succeeds:
```
west build -b reel_board samples/hello_world --sysbuild -- -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DSB_CONFIG_BOOT_ENCRYPTION=y
```

In v4.4.0-rc3 it fails to compile with missing `mbedtls/aes.h` header:
```
In file included from /home/aaron/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/crypto/aes_ctr.h:21,
                 from /home/aaron/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/enc_key.h:33,
                 from /home/aaron/zephyrproject/bootloader/mcuboot/boot/bootutil/src/image_validate.c:54:
/home/aaron/zephyrproject/bootloader/mcuboot/boot/bootutil/include/bootutil/crypto/aes_ctr_mbedtls.h:17:10: fatal error: mbedtls/aes.h: No such file or directory
   17 | #include <mbedtls/aes.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
```

git bisect shows commit 66ee2d2304eea0c1fa4508763f0b978d6681cea0 where it fails to build, but that was for an unrelated header file that may have been fixed.


### Regression

- [x] This is a regression.

### Steps to reproduce

west build -b reel_board samples/hello_world --sysbuild -- -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DSB_CONFIG_BOOT_ENCRYPTION=y


### Relevant log output

```shell

```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

v4.4.0-rc3
sdk 1.0.0


### Additional Context

_No response_

--- a/modules/mbedtls/legacy_support.cmake
+++ b/modules/mbedtls/legacy_support.cmake
@@ -17,6 +17,7 @@ if(CONFIG_MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)
   )
   set(MBEDTLS_PRIVATE_INCLUDE_PATH "${ZEPHYR_TF_PSA_CRYPTO_MODULE_DIR}/drivers/builtin/include/mbedtls/private")
   set(legacy_headers
+    ${MBEDTLS_PRIVATE_INCLUDE_PATH}/aes.h
     ${MBEDTLS_PRIVATE_INCLUDE_PATH}/bignum.h
     ${MBEDTLS_PRIVATE_INCLUDE_PATH}/cipher.h
     ${MBEDTLS_PRIVATE_INCLUDE_PATH}/cmac.h
@@ -28,6 +29,13 @@ if(CONFIG_MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)
     ${MBEDTLS_PRIVATE_INCLUDE_PATH}/rsa.h
   )
   file(COPY ${legacy_headers} DESTINATION ${CMAKE_BINARY_DIR}/legacy-mbedtls-headers/mbedtls/)
+  if(CONFIG_MCUBOOT)
+    set(MBEDTLS_BUILTIN_SRC_PATH "${ZEPHYR_TF_PSA_CRYPTO_MODULE_DIR}/drivers/builtin/src")
+    set(legacy_headers
+      ${MBEDTLS_BUILTIN_SRC_PATH}/rsa_alt_helpers.h
+    )
+    file(COPY ${legacy_headers} DESTINATION ${CMAKE_BINARY_DIR}/legacy-mbedtls-headers/)
+  endif()
   target_include_directories(mbedTLS INTERFACE
     ${CMAKE_BINARY_DIR}/legacy-mbedtls-headers/
   )

--- a/west.yml
+++ b/west.yml
@@ -335,7 +335,7 @@ manifest:
       groups:
         - crypto
     - name: mcuboot
-      revision: 2d42e8eadcdb76737f99a0e03e6c6e322f8f1639
+      revision: ee39e2d694bd827ffd1bebbce2f571a9154e6ec2
       path: bootloader/mcuboot
       groups:
         - bootloader

